program-runtime/src/loaded_programs.rs

```diff
@@ -420,7 +427,10 @@ impl LoadedPrograms {
             if existing.deployment_slot == entry.deployment_slot
                 && existing.effective_slot == entry.effective_slot
             {
-                if matches!(existing.program, LoadedProgramType::Unloaded) {
+                if matches!(existing.program, LoadedProgramType::Builtin(_)) {
```

Comment:
Could an existing builtin be replaced with a new one that's deployed/effective in the same slot?

Reply:
Yes, surprised me as well, seems to be intended behavior. There even is a test for it:
https://github.com/solana-labs/solana/blob/e84613b54ce2ea1b72fb8eef593361dde6a5fcf5/runtime/src/bank/tests.rs#L4861

Reply:
I'll take a look in a follow on PR.